File: recursion/s9-problems-2/exercise-15/solution.js
/*
Write a recursive function called reverse which accepts a string and returns a new string in reverse
*/

function reverse(s) {
  if (s.length === 1) return s;
  return s.slice(-1) + reverse(s.slice(0, -1));
}

console.log(reverse("awesome")); // 'emosewa'
console.log(reverse("rithmschool")); // 'loohcsmhtir'

File: recursion/s9-problems-2/exercise-15/solution.go
package main

import "fmt"

func main() {
	fmt.Println(reverse("awesome"))
}

func reverse(s string) string {
	if len(s) == 1 {
		return s
	}
	return string(s[len(s)-1]) + reverse(s[:len(s)-1])
}
